src/main/java/com/qunite/api/security/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,54 @@
+package com.qunite.api.security;
+
+import static com.qunite.api.security.JwtConstants.ROLES;
+import static com.qunite.api.security.JwtConstants.TOKEN_PREFIX;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Set;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthorizationFilter extends OncePerRequestFilter {
+  private final JwtService jwtService;
+
+  @Override
+  protected void doFilterInternal(@NonNull HttpServletRequest request,
+                                  @NonNull HttpServletResponse response,
+                                  @NonNull FilterChain filterChain)
+      throws ServletException, IOException {
+    final String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+    if (!this.isHeaderValid(authHeader)) {
+      filterChain.doFilter(request, response);
+      return;
+    }
+    jwtService.verifyAccessToken(authHeader.substring(7))
+        .ifPresent(decodedJWT -> {
+          var username = decodedJWT.getSubject();
+          var authToken = new UsernamePasswordAuthenticationToken(
+              username, null, Set.of(new SimpleGrantedAuthority(ROLES))
+          );
+          SecurityContextHolder.getContext().setAuthentication(authToken);
+        });
+    filterChain.doFilter(request, response);
+  }
+
+  private boolean isHeaderValid(String authHeader) {
+    return Objects.nonNull(authHeader) && authHeader.startsWith(TOKEN_PREFIX);
```

Comment:
I'm not sure, but nonNull can throw an error if it's null, I recommend using the usual !=null